File: rxbindroid/src/main/java/moe/pine/rx/bindroid/RxBindroid.java
package moe.pine.rx.bindroid;

import android.app.Activity;
import android.app.Fragment;
import android.view.View;

import moe.pine.rx.bindroid.app.AppObservable;
import moe.pine.rx.bindroid.view.ViewObservable;
import rx.Observable;

/**
 * RxBindroid
 * Created by pine on 2016/03/06.
 */
public class RxBindroid {
    public static <T> Observable<T> bindActivity(Activity activity, Observable<T> observable) {
        return AppObservable.bindActivity(activity, observable);
    }

    public static <T> Observable<T> bindFragment(Fragment fragment, Observable<T> observable) {
        return AppObservable.bindFragment(fragment, observable);
    }

    public static <T> Observable<T> bindFragment(android.support.v4.app.Fragment fragment, Observable<T> observable) {
        return AppObservable.bindSupportFragment(fragment, observable);
    }

    public static <T> Observable<T> bindView(View view, Observable<T> observable) {
        return ViewObservable.bindView(view, observable);
    }
}

File: rxbindroid-kotlin/src/main/kotlin/moe/pine/rx/bindroid/RxBindroid.kt
package moe.pine.rx.bindroid

import android.app.Activity
import android.app.Fragment
import android.view.View
import rx.Observable

/**
 * RxBindroid Extensions for Observable
 * Created by pine on 2016/03/06.
 */

fun <T> Observable<T>.bindActivity(activity: Activity): Observable<T> {
    return RxBindroid.bindActivity(activity, this)
}

fun <T> Observable<T>.bindFragment(fragment: Fragment): Observable<T> {
    return RxBindroid.bindFragment(fragment, this)
}

fun <T> Observable<T>.bindFragment(fragment: android.support.v4.app.Fragment): Observable<T> {
    return RxBindroid.bindFragment(fragment, this)
}

fun <T> Observable<T>.bindView(view: View): Observable<T> {
    return RxBindroid.bindView(view, this)
}